autokeras/text/text_preprocessor.py

```diff
@@ -139,12 +139,15 @@ def processing(path, word_index, input_length, x_train):
     embedding_matrix = load_pretrain(path=path, word_index=word_index)
 
     # Get the first available GPU
-    device_id_list = GPUtil.getFirstAvailable()
-    device_id = device_id_list[0]  # grab first element from list
+    try:
+        device_id_list = GPUtil.getFirstAvailable()
+        device_id = device_id_list[0]  # grab first element from list
+        # Set CUDA_VISIBLE_DEVICES to mask out all other GPUs than the first available device id
+        os.environ["CUDA_VISIBLE_DEVICES"] = str(device_id)
+        device = '/gpu:0'
```

Comment:
You may need to discuss with Lin to see how to get the device. We need to call some function instead of directly use the env var.